### DISCUSSION TITLE
Internal Server Error

### DISCUSSION THREAD

**Discussion by LordSith77** [seerr-team/seerr#801] [Q&A] [answered]
I've had Jellyseerr installed and running great for a few months now. A couple days ago, I was alerted that it was not showing up for users anymore, and instead had an error page. I couldn't find the cause, so I removed Jellyseerr completely, then re-installed it (using the guide on the main page (linked to the docker setup). This morning, it was erroring again, so I reset it once again. Now a few hours later, it's  having an internal server error again.. I get it all set up, scan the Jellyfin movies and TV, and everything is working perfectly. Then a few hours later, it's all gone again. 
![image](https://github.com/Fallenbagel/jellyseerr/assets/26206442/c099d6f8-6684-4286-9539-cb706dce3400)

I have it running in a reverse proxy (Proxy redirects to :5055)
I've checked all my Apache settings, and everything is exactly the same as it was when it was working. 
As far as I can tell, I'm thinking it's the:
<code>volumes:
            - /path/to/appdata/config:/app/config</code>
  So this time, I followed another sites directions, and going by what they said, I changed that line to:
<code>~/compose-jellyseerr/config:/app/config</code>
This folder is located in my user home director (Home/jim/compose-jellyseer)
![image](https://github.com/Fallenbagel/jellyseerr/assets/26206442/8ee8a61b-8a02-4efd-9035-cc1b2a5e37e7)
![image](https://github.com/Fallenbagel/jellyseerr/assets/26206442/9301746c-4764-4bda-96df-780925bc5d29)
I'm at a complete loss at how to fix this. Any help would be greatly appreciated. 

System info:
![image](https://github.com/Fallenbagel/jellyseerr/assets/26206442/f7f556b2-4c17-45f7-aa92-0a29ace097a5)

**Comment by fallenbagel** [comment]
Please hop onto discord for troubleshooting.

Also /path/to/config is a placeholder not a real path. And the error you're facing is an apache error not jellyseerr. Unless you can't access it on lan too

**Comment by LordSith77** [comment]
I thought it was an apache issue, but everything looked normal, so it was really confusing me. Then I couldn't access it on LAN either, so figured it was Jellyseerr. However, I just rebooted my entire server, and weirdly, it's working again. I'm so confused now. LOL

So the path/to/config doesn't matter?

But if I have any more issues, I'll hop on the Discord. Thanks for the response, and for the program itself. I love it, and it allows all my users to request...everyone of my users loves this. :-)

**Reply by fallenbagel** [reply]
> So the path/to/config doesn't matter?
> 
No it does but you need to map it with a proper host folder or docker volume. As it is, it's not a real path

**Reply by LordSith77** [reply]
Ok, so I did install it using docker (first time ever using docker, so I'm not that familiar with it, so please bear with me.) and if my understanding is correct, I need to point the path to the var/lib/docker location, correct? 
So here:
![image](https://github.com/Fallenbagel/jellyseerr/assets/26206442/ffd59bcd-95c1-4383-83ed-275e618f4e96)

**Reply by fallenbagel** [reply]
> Ok, so I did install it using docker (first time ever using docker, so I'm not that familiar with it, so please bear with me.) and if my understanding is correct, I need to point the path to the var/lib/docker location, correct? 
> So here:
> ![image](https://github.com/Fallenbagel/jellyseerr/assets/26206442/ffd59bcd-95c1-4383-83ed-275e618f4e96)
> 

No.
You need to map it somewhere. Proper. If not the minute you recreate your container everything will be gone.

```
volumes:
    - /opt/jellyseerr:/app/config
```
For example ^^
And it has to have proper permissions. If you dont want to deal with permissions you can create a docker volume

`docker volume create jellyseerr_config`

Then 

```yaml
version: '3'
services:
    jellyseerr:
       image: fallenbagel/jellyseerr:latest
       container_name: jellyseerr
       environment:
            - LOG_LEVEL=debug
            - TZ=Asia/Tashkent
       ports:
            - 5055:5055
       volumes:
            - jellyseerr_config:/app/config
       restart: unless-stopped 

volumes:
     jellyseerr_config:
        external: true
```

**Comment by LordSith77** [comment]
Ah, ok. Hmm, interesting. When I went in to edit it, it was already changed
to:
    volumes:
      - ./config:/app/config
And it's in home/jim/dockers/jellyseerr

**Reply by fallenbagel** [reply]
As long as the perms are correct that's fine too. You can make sure if perms are correct by checking your settings.json and stuff and see if your server details are saved. If so you're good :D

**Comment by LordSith77** [comment]
Awesome, thank you so much! Inside the settings.json shows both URL's to
Jellyfin and Jellyseerr.

But I just discovered I have two separate folders each with
docker-compose.yml and folders for app and config. Inside the config of
both is the settings.json. Now I'm not sure which is the one it's using. Is
there a way to find out, so I can delete the other one?
home/jim/compose-jellyseer (this one in settings only shows the URL for
Jellyfin)
home/jim/dockers/jellyyseer (this one in the settings shows both URL's)

**Reply by fallenbagel** [reply]
> Awesome, thank you so much! Inside the settings.json shows both URL's to
> Jellyfin and Jellyseerr.
> 
> But I just discovered I have two separate folders each with
> docker-compose.yml and folders for app and config. Inside the config of
> both is the settings.json. Now I'm not sure which is the one it's using. Is
> there a way to find out, so I can delete the other one?
> home/jim/compose-jellyseer (this one in settings only shows the URL for
> Jellyfin)
> home/jim/dockers/jellyyseer (this one in the settings shows both URL's)
> 
> On Fri, Jun 7, 2024 at 3:14 PM Fallenbagel ***@***.***> wrote:
> 
> > As long as the perms are correct that's fine too. You can make sure if
> > perms are correct by checking your settings.json and stuff and see if your
> > server details are saved. If so you're good :D
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <https://github.com/Fallenbagel/jellyseerr/discussions/801#discussioncomment-9707026>,
> > or unsubscribe
> > <https://github.com/notifications/unsubscribe-auth/AGH6B2XFTD4PDSSDFRH4IWTZGIIB7AVCNFSM6AAAAABI64MOZSVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TOMBXGAZDM>
> > .
> > You are receiving this because you authored the thread.Message ID:
> > ***@***.***>
> >
> 

Check the apikey. See which one matches the one that's shown on your jellyseerr web ui

**Comment by LordSith77** [comment]
You're not going to believe this.... neither of those locations had the
right API key. So I did a search of my system, and found the settings.json
file with the correct key. Located at:
/path/to/appdata/config/settings.json
So it apparently made those folders. Will it work this way, or how can I
move it to where it should be?

**Reply by fallenbagel** [reply]
Not a good idea to use that. Permission nightmare.

Just stop jellyseerr and docker rm jellyseerr and use one of the other docker compose

**Comment by LordSith77** [comment]
Yeah, I had a feeling. So is there an easy way to move the config folder to
where it should be?

**Reply by fallenbagel** [reply]
You just map the config folder on the compose

**Comment by LordSith77** [comment, marked as answer]
I think I finally got it set right. I just removed everything and started
over (since I had to do that a couple days ago, there wasn't much in the
requests...and what was already in there, I've bookmarked the IMDB pages),
and now I'm pretty sure I have it all set up properly. I followed the
instructions from this page:
https://www.llewellynhughes.co.uk/post/installing-jellyseerr/
And it worked perfectly. Thank you for all your help in understanding what
I did wrong. LOL I've learned a lot more about dockers now, so thanks for
that as well. :-)

**Comment by doktoil-makresh** [comment]
Hello @LordSith77 can you share your Apache reverse proxy configuration?

**Reply by LordSith77** [reply]
I use VirtualMin (As I host my on websites), and just created a subdomain, and enable proxy, and pointed it to my subdomain name with the :5055 and it works great.
But here's the proxy section of my .conf file:

`RedirectMatch ^/(?!.well-known)(.*)$ https://jellyseerr.lordsith77.com/$1

IPCCommTimeout 3001

ProxyPass / http://jellyseerr.lordsith77.com:5055/

ProxyPassReverse / http://jellyseerr.lordsith77.com:5055/

SSLProxyEngine on

RewriteCond %{HTTPS} off

RewriteRule ^/(?!.well-known)(.*)$ https://%{HTTP_HOST}/$1 [R]`

**Comment by doktoil-makresh** [comment]
Thank you @LordSith77 . Unfortunately, I'm looking for a working configuration where jellyseerr is in a basedir (my.domain/jellyseerr). I'll restart working on it as this is now supported by the latest Jellyseerr version.

**Reply by fallenbagel** [reply]
> Thank you @LordSith77 . Unfortunately, I'm looking for a working configuration where jellyseerr is in a basedir (my.domain/jellyseerr). I'll restart working on it as this is now supported by the latest Jellyseerr version.

No it is not. Base directory is not officially supported on any version

**Reply by doktoil-makresh** [reply]
@Fallenbagel I may have misunderstood the docs, as it is possible to define the application URL and enable reverse proxy support. Thank you for your support and your great work on this project

**Reply by fallenbagel** [reply]
> @Fallenbagel I may have misunderstood the docs, as it is possible to define the application URL and enable reverse proxy support. Thank you for your support and your great work on this project

Application URL is used in stuff like emails and stuff for links. Reverse proxy enabling makes it so that it sends the forward proxy so you get the client ip from the reverse proxy instead of 127.0.0.1
